### PR TITLE
Cleanup conditional-bypass query

### PR DESCRIPTION
**Pull request by max-schaefer**
A bit of clean-up for the experimental query contributed by @porcupineyhairs  in #121.

Commit-by-commit review makes the changes easier to follow.

### CODE REVIEW

File: ql/src/experimental/CWE-840/ConditionalBypass.qhelp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+The following example shows a comparison where both the sides are from attacker-controlled request
+headers. This should be avoided:
+        </p>
+        <sample src="ConditionalBypassBad.go" />
+        <p>
+One way to remedy the problem is to test against a value stored in a configuration:
+        </p>
+        <sample src="ConditionalBypassGood.go" />
+    </example>
+</qhelp>
```
Comment (porcupineyhairs) [inline ql/src/experimental/CWE-840/ConditionalBypass.qhelp:26]: I think a new line is required here
```suggestion
</qhelp>

```

File: ql/src/experimental/CWE-840/ConditionalBypass.ql [hunk 2]
```diff
@@ -11,10 +11,10 @@
 import go
 
 /**
- * A data-flow configuration for reasoning about Condition Bypass.
+ * A taint-tracking configuration for reasoning about conditional bypass.
  */
 class Configuration extends TaintTracking::Configuration {
-  Configuration() { this = "Comparision Expression Check Bypass" }
+  Configuration() { this = "ConitionalBypass" }
```
Comment (porcupineyhairs) [inline ql/src/experimental/CWE-840/ConditionalBypass.ql:17]: Typo. I think this should be
```suggestion
  Configuration() { this = "ConditionalBypass" }
```

File: ql/src/experimental/CWE-840/ConditionalBypass.ql [hunk 3]
```diff
@@ -35,5 +35,6 @@ where
   rhs.getNode().asExpr() = c.getRightOperand() and
   config.hasFlowPath(lhsSource, lhs) and
   lhs.getNode().asExpr() = c.getLeftOperand()
-select c, "This comparision is between user controlled operands derived from $@", lhsSource,
-  " and $@", rhsSource, "hence may be bypassed."
+select c,
+  "This comparison compares user-controlled values from $@ and $@, and hence can be bypassed.",
+  lhsSource, "here", rhsSource, "here"
```
Comment (porcupineyhairs) [inline ql/src/experimental/CWE-840/ConditionalBypass.ql:40]: I am not sure I understand this. How will `"here"` be displayed here? Shouldn't this be something like:
```suggestion
  lhsSource, rhsSource
```
the lhsSource and rhsSource will replace the two `$@`s.

Comment (max-schaefer) [reply ql/src/experimental/CWE-840/ConditionalBypass.ql:40]: No. Please refer to [the documentation](https://help.semmle.com/QL/learn-ql/writing-queries/select-statement.html) on how to use $@.

File: ql/src/experimental/CWE-840/ConditionalBypass.qhelp (hunk 1 above)
Comment (max-schaefer) [reply ql/src/experimental/CWE-840/ConditionalBypass.qhelp:26]: I don't think so.

File: ql/src/experimental/CWE-840/ConditionalBypass.ql (hunk 2 above)
Comment (max-schaefer) [reply ql/src/experimental/CWE-840/ConditionalBypass.ql:17]: Thanks!

**Comment by max-schaefer** [comment]
@sauyon *prod*

File: ql/src/experimental/CWE-840/ConditionalBypass.ql [hunk 4]
```diff
@@ -1,8 +1,8 @@
 /**
- * @name Comparision Expression Check Bypass
- * @description Comparing two user controlled inputs may
- * lead to an effective bypass of the comparison check.
- * @id go/condition-bypass
+ * @name User-controlled bypass of condition
+ * @description A check that compares two user-controlled inputs with each other can be bypassed
+ *              by a malicious user.
```
Comment (sauyon) [inline ql/src/experimental/CWE-840/ConditionalBypass.ql:4]:
```suggestion
 * by a malicious user.
```

Were we removing these?

File: ql/src/experimental/CWE-840/ConditionalBypass.qhelp [hunk 5]
```diff
@@ -2,26 +2,25 @@
 <qhelp>
     <overview>
         <p>
```
Comment (sauyon) [inline ql/src/experimental/CWE-840/ConditionalBypass.qhelp:4]: I guess we don't have a style for qhelp files?

File: ql/src/experimental/CWE-840/ConditionalBypass.ql (hunk 4 above)
Comment (max-schaefer) [reply ql/src/experimental/CWE-840/ConditionalBypass.ql:4]: Looking through our existing queries, it seems like we generally indent the second line.

File: ql/src/experimental/CWE-840/ConditionalBypass.qhelp (hunk 5 above)
Comment (max-schaefer) [reply ql/src/experimental/CWE-840/ConditionalBypass.qhelp:4]: I don't think we do.

File: ql/src/experimental/CWE-840/ConditionalBypass.ql (hunk 4 above)
Comment (sauyon) [reply ql/src/experimental/CWE-840/ConditionalBypass.ql:4]: Ah, indeed.

**Pull request merged**